### PR TITLE
chore: Add additional debug logs for session functions

### PR DESCRIPTION
**Pull request by nkshah2**
## Summary of change

- Adds debug logs whenever the SDK returns try refresh token or unauthorized errors to make debugging easier

## Related issues

- 

## Documentation changes
NA

## Checklist for important updates

-   [x] Changelog has been updated
-   [ ] `coreDriverInterfaceSupported.json` file has been updated (if needed)
    -   Along with the associated array in `supertokens/constants.go`
-   [ ] `frontendDriverInterfaceSupported.json` file has been updated (if needed)
-   [ ] Changes to the version if needed
    -   In `supertokens/constants.go > version variable`
-   [x] Had installed and ran the pre-commit hook
-   [x] Issue this PR against the latest non released version branch.
    -   To know which one it is, run find the latest released tag (`git tag`) in the format `vX.Y.Z`, and then find the latest branch (`git branch --all`) whose `X.Y` is greater than the latest released tag.
    -   If no such branch exists, then create one from the latest released branch.

## Remaining TODOs for this PR

-   [ ] ...

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [unreleased]
```
Comment (rishabhpoddar) [inline CHANGELOG.md:9]: increase the patch version

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/session/accessToken.go [hunk 2]
```diff
@@ -81,6 +84,7 @@ func GetInfoFromAccessToken(jwtInfo sessmodels.ParsedJWTInfo, jwks keyfunc.JWKS,
 			}
 
 			if parseErr != nil {
+				supertokens.LogDebugMessage("GetInfoFromAccessToken: Returning TryRefreshTokenError because access token parsing failed")
```
Comment (rishabhpoddar) [inline recipe/session/accessToken.go:87]: add the error message itself in the log. That will be `parseErr.Error()` right?

File: recipe/session/accessToken.go [hunk 3]
```diff
@@ -107,13 +112,15 @@ func GetInfoFromAccessToken(jwtInfo sessmodels.ParsedJWTInfo, jwks keyfunc.JWKS,
...
 	if payload == nil {
+		supertokens.LogDebugMessage("GetInfoFromAccessToken: Returning TryRefreshTokenError because access token JWT has no payload")
 		return nil, sterrors.TryRefreshTokenError{
 			Msg: "Invalid JWT",
 		}
 	}
 
 	err := ValidateAccessTokenStructure(payload, jwtInfo.Version)
 	if err != nil {
+		supertokens.LogDebugMessage("GetInfoFromAccessToken: Returning TryRefreshTokenError because ValidateAccessTokenStructure returned an error")
```
Comment (rishabhpoddar) [inline recipe/session/accessToken.go:123]: what is the error though?

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/session/sessionFunctions.go [hunk 4]
```diff
@@ -234,17 +236,20 @@ func refreshSessionHelper(config sessmodels.TypeNormalisedInput, querier superto
 
 	response, err := querier.SendPostRequest("/recipe/session/refresh", requestBody)
 	if err != nil {
+		supertokens.LogDebugMessage("refreshSessionHelper: Call to /recipe/session/refresh API failed")
```
Comment (rishabhpoddar) [inline recipe/session/sessionFunctions.go:239]:
```suggestion
```

File: recipe/session/sessionFunctions.go [hunk 5]
```diff
@@ -234,17 +236,20 @@ func refreshSessionHelper(config sessmodels.TypeNormalisedInput, querier superto
...
 		return sessmodels.CreateOrRefreshAPIResponse{}, err
 	}
 	if response["status"] == "OK" {
 		delete(response, "status")
 		responseByte, err := json.Marshal(response)
 		if err != nil {
+			supertokens.LogDebugMessage("refreshSessionHelper: Could not parse response from /recipe/session/refresh API")
```
Comment (rishabhpoddar) [inline recipe/session/sessionFunctions.go:246]:
```suggestion
```

File: recipe/session/sessionFunctions.go [hunk 6]
```diff
@@ -234,17 +236,20 @@ func refreshSessionHelper(config sessmodels.TypeNormalisedInput, querier superto
...
 			return sessmodels.CreateOrRefreshAPIResponse{}, err
 		}
 		var result sessmodels.CreateOrRefreshAPIResponse
 		err = json.Unmarshal(responseByte, &result)
 		if err != nil {
+			supertokens.LogDebugMessage("refreshSessionHelper: Could not decode response from /recipe/session/refresh API")
```
Comment (rishabhpoddar) [inline recipe/session/sessionFunctions.go:252]:
```suggestion
```

**Review by rishabhpoddar — CHANGES_REQUESTED** [review]

File: recipe/session/accessToken.go (hunk 3 above)
Comment (nkshah2) [reply recipe/session/accessToken.go:124]: ValidateAccessTokenStructure logs why it threw an error

**Pull request merged**